### DISCUSSION TITLE
Please Help: Batch Process a Text File to Image

### DISCUSSION THREAD

**Discussion by Konsider** [ImageMagick/ImageMagick#4737] [General] [answered]
Hi 😊

I am a complete newbie, but I wonder if someone would be kind enough to help me?

I am looking to take a text file, and turn each line into an image (png) with a transparent background.

I would like to specify parameters such as image size, text font, text size, etc.

The image would be named the same as the text. For example, a text file with:

This
Is
A
Test

Would create four images, one with each word, each named the same as the word.

I am on Windows and have installed: ImageMagick-7.1.0-19-Q16-HDRI-x64-dll.exe

Thank you for any help! :)

**Comment by snibgo** [comment]
Assuming your text file is called textfile.txt, here is a BAT script:
```
for /F "tokens=*" %%L in (textfile.txt) do magick ^
  -pointsize 50 ^
  -background None ^
  caption:"%%L" ^
  "%%L.png"
```

**Reply by Konsider** [reply]
That's brilliant! Thanks so much! :) I wonder if you might be able to tell me the lines to add to the batch file in order to:

specify a uniform image size (the image currently matches to the word length)
choose a font
choose the text colour
choose word formatting (currently it seems to always do italic?)

Thanks a lot! :)

**Comment by snibgo** [comment, marked as answer]
See the options in https://imagemagick.org/script/command-line-options.php for "-size", "-font", "-fill" and so on.

**Reply by Konsider** [reply]
Perfect! Thank you for taking the time to help me - I appreciate it! :)

**Reply by Konsider** [reply]
> See the options in https://imagemagick.org/script/command-line-options.php for "-size", "-font", "-fill" and so on.

Sorry, but I am stuck. I am trying to offset the text. The gravity is center, but I want to push the text down by a few pixels. I can't seem to work out how to do it from the options?

**Reply by fmw42** [reply]
You could add a blank row at the top and or remove a row at the bottom using -splice and -chop. See https://legacy.imagemagick.org/Usage/crop/#splice

**Reply by Konsider** [reply]
That sounded like a great idea, but I am not sure it works. I get the error "no images found for operation splice". Here is my complete batch file:

`for /F "tokens=*" %%L in (words.txt) do magick ^
  -pointsize 100 ^
  -size 800x800 ^
  -fill yellow ^
  -gravity center ^
  -style normal ^
  -font Algerian ^
  -background None ^
  -splice 0x100 ^
  caption:"%%L" ^
  "output\%%L.png"`

**Reply by fmw42** [reply]
You have to splice or chop after the caption:...  and before the output.  The caption has to create the image before you can splice or chop on it.

**Reply by fmw42** [reply]
You can also just change -gravity center to -gravity south.

**Reply by Konsider** [reply]
Man, I feel like I am so close to getting it, but every combination of splice parameters I try, it just cuts the text up without moving it in relation to the background :( (I now have it after the caption, so at least splice is working now, thanks :) )

Also, the problem with gravity south is, it puts the text directly flush to the base of the image. If we could use that, but "pad up" the text so it was some pixels from the bottom, that would be fine :)

**Reply by Konsider** [reply]
GOT IT!!! I had to make the image size smaller by the amount I will be splicing! :) Thanks a lot for all of your help, that is awesome!
